src/main/kotlin/com/github/sanity/kweb/KWeb.kt

```diff
@@ -34,19 +43,20 @@ class KWeb(val port: Int,
            val debug: Boolean = false,
            val refreshPageOnHotswap : Boolean = false,
            val plugins: List<KWebPlugin> = Collections.emptyList(),
-           val appServerConfigurator: (AppServer) -> Unit = {},
+           // val appServerConfigurator: (AppServer) -> Unit = {},
            val onError : ((List<StackTraceElement>, JavaScriptError) -> LogError) = { _, _ ->  true},
            val buildPage: RootReceiver.() -> Unit
 ) {
     companion object: KLogging()
 
-    private val server = AppServer(AppConfiguration(port = port))
+    private var server: NettyApplicationHost? = null
     private val clients: MutableMap<String, WSClientData>
     private val mutableAppliedPlugins: MutableSet<KWebPlugin> = HashSet()
     val appliedPlugins: Set<KWebPlugin> get() = mutableAppliedPlugins
 
     init {
-        appServerConfigurator.invoke(server)
+        // TODO How is this used?
+        // appServerConfigurator.invoke(server)
```

Comment:
Just to answer your question directly, it is used to add additional routes and route handlers, for example it is used by the [select2](https://select2.github.io/) plugin to create an endpoint so that select2 can request suggestions from the server in a nice convenient way.